bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschSHCBridgeHandler.java

```diff
@@ -41,81 +31,58 @@
 
 public class BoschSHCBridgeHandler extends BaseBridgeHandler {
 
-    public BoschSHCBridgeHandler(Bridge bridge) {
+
+public BoschSHCBridgeHandler(Bridge bridge) {
         super(bridge);
     }
 
     private final Logger logger = LoggerFactory.getLogger(BoschSHCBridgeHandler.class);
 
-    private @Nullable HttpClient httpClient;
+    private @Nullable BoschHttpClient httpClient;
 
     private @Nullable ArrayList<Room> rooms;
     private @Nullable ArrayList<Device> devices;
 
     private @Nullable String subscriptionId;
 
-    private SslContextFactory getSslContext(String keystore, String keystorePassword) {
-
-        // Instantiate and configure the SslContextFactory
-        // SslContextFactory.Client sslContextFactory = new SslContextFactory.Client();
-        SslContextFactory sslContextFactory = new SslContextFactory(true); // Accept all certificates
-
-        // Keystore for managing the keys that have been used to pair with the SHC
-        // https://www.eclipse.org/jetty/javadoc/9.4.12.v20180830/org/eclipse/jetty/util/ssl/SslContextFactory.html
-        sslContextFactory.setKeyStorePath(keystore);
-        sslContextFactory.setKeyStorePassword(keystorePassword);
 
-        // Bosch is using a self signed certificate
-        sslContextFactory.setTrustAll(true);
-        sslContextFactory.setValidateCerts(false);
-        sslContextFactory.setValidatePeerCerts(false);
-        sslContextFactory.setEndpointIdentificationAlgorithm(null);
-
-        return sslContextFactory;
-    }
 
-    // https://stackoverflow.com/questions/13894699/java-how-to-store-a-key-in-keystore
-    public X509Certificate generateCertificate(KeyPair keyPair) {
-
-        return null;
-    }
 
     @Override
     public void initialize() {
 
         config = getConfigAs(BoschSHCBridgeConfiguration.class);
+        
+        // TODO if we find a good place for a default JKS keystore file, we can skip the config.keystorePath.
+        
+        // use the Bosch SHC password for keystore if no own password given
+        if (config.keystorePassword == null | config.keystorePassword.isEmpty()) {
```

Comment:
You have to use two ||, otherwise you'll get a NullReferenceException when the password is not set because the second part of the check is executed even if the keystorePassword is null.

Reply:
Wow, well spotted :-)

Reply:
Thanks a lot! Fixed

Reply:
Pushed new rebased commits including this fix. Not sure if this was the right way to resolve the conflicts.